plugins/database/cassandra/connection_producer.go

```diff
@@ -239,7 +240,15 @@ func (c *cassandraConnectionProducer) createSession() (*gocql.Session, error) {
 
 	// Verify the info
 	err = session.Query(`LIST ALL`).Exec()
-	if err != nil {
+	if err != nil && strings.Contains(err.Error(), "not authorized") {
```

Comment:
Can you make sure the username is not empty too?

Reply:
Can do, but since the init checks it, isn't checking again a bit redundant?